src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -3280,35 +3284,44 @@ public ResultSet getSchemas(String originalCatalog, String originalSchema) throw
     raiseSQLExceptionIfConnectionIsClosed();
 
     // apply session context when catalog is unspecified
-    SFPair<String, String> resPair = applySessionContext(originalCatalog, originalSchema);
-    final String catalog = resPair.left;
-    final String schemaPattern = resPair.right;
+    ContextAwareMetadataSearch result = applySessionContext(originalCatalog, originalSchema);
+    final String catalog = result.database();
+    final String schemaPattern = result.schema();
+    boolean isExactSchema = result.isExactSchema();
 
     final Pattern compiledSchemaPattern = Wildcard.toRegexPattern(schemaPattern, true);
 
-    String showSchemas = "show /* JDBC:DatabaseMetaData.getSchemas() */ schemas";
+    StringBuilder showSchemas =
+        new StringBuilder("show /* JDBC:DatabaseMetaData.getSchemas() */ schemas");
 
     Statement statement = connection.createStatement();
-    // only add pattern if it is not empty and not matching all character.
-    if (schemaPattern != null
+    if (isExactSchema) {
+      String escapedSchema =
+          schemaPattern.replaceAll("_", "\\\\\\\\_").replaceAll("%", "\\\\\\\\%");
```

Comment:
Can you help me understand this? 
Why so many backslashes?
Would it make sense to introduce an escape function, similarly to `escapeSingleQuoteForLikeCommand`?